heartbeat/sender/multikeyHeartbeatSender.go

```diff
@@ -0,0 +1,209 @@
+package sender
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/ElrondNetwork/covalent-indexer-go/process"
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/ElrondNetwork/elrond-go-core/core/check"
+	"github.com/ElrondNetwork/elrond-go/heartbeat"
+)
+
+// argMultikeyHeartbeatSender represents the arguments for the heartbeat sender
+type argMultikeyHeartbeatSender struct {
+	argBaseSender
+	peerTypeProvider     heartbeat.PeerTypeProviderHandler
+	versionNumber        string
+	baseVersionNumber    string
+	nodeDisplayName      string
+	identity             string
+	peerSubType          core.P2PPeerSubType
+	currentBlockProvider heartbeat.CurrentBlockProvider
+	keysHolder           heartbeat.KeysHolder
+	shardCoordinator     process.ShardCoordinator
+}
+
+type multikeyHeartbeatSender struct {
+	commonHeartbeatSender
+	peerTypeProvider  heartbeat.PeerTypeProviderHandler
+	versionNumber     string
+	baseVersionNumber string
+	nodeDisplayName   string
+	identity          string
+	peerSubType       core.P2PPeerSubType
+	keysHolder        heartbeat.KeysHolder
+	shardCoordinator  process.ShardCoordinator
+}
+
+// newMultikeyHeartbeatSender creates a new instance of type multikeyHeartbeatSender
+func newMultikeyHeartbeatSender(args argMultikeyHeartbeatSender) (*multikeyHeartbeatSender, error) {
+	err := checkMultikeyHeartbeatSenderArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &multikeyHeartbeatSender{
+		commonHeartbeatSender: commonHeartbeatSender{
+			baseSender:           createBaseSender(args.argBaseSender),
+			currentBlockProvider: args.currentBlockProvider,
+		},
+		versionNumber:     args.versionNumber,
+		baseVersionNumber: args.baseVersionNumber,
+		nodeDisplayName:   args.nodeDisplayName,
+		identity:          args.identity,
+		peerSubType:       args.peerSubType,
+		peerTypeProvider:  args.peerTypeProvider,
+		keysHolder:        args.keysHolder,
+		shardCoordinator:  args.shardCoordinator,
+	}, nil
+}
+
+func checkMultikeyHeartbeatSenderArgs(args argMultikeyHeartbeatSender) error {
+	err := checkBaseSenderArgs(args.argBaseSender)
+	if err != nil {
+		return err
+	}
+	if check.IfNil(args.peerTypeProvider) {
+		return heartbeat.ErrNilPeerTypeProvider
+	}
+	if len(args.versionNumber) > maxSizeInBytes {
```

Comment:
baseVersionNumber and peerSubType shouldn't be checked for some required len/values ?

Reply:
pushed check for `baseVersionNumber`.. `peerSubType` is declared as `type P2PPeerSubType uint32` so no check needed